### PR TITLE
Fix callback error reporting

### PR DESCRIPTION
**Pull request by alexcjohnson**
#753 simplified the base `Component` class to prohibit fewer prop names - but the `keys` method was actually being used, leading to #791. Easy to replace it with a list comprehension (which calls `__iter__` under the hood).

- [x] I have run the tests locally and they passed. (refer to testing section in [contributing](https://github.com/plotly/dash/blob/master/CONTRIBUTING.md))
- [x] I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR
- [x] I have added entry in the `CHANGELOG.md`

### CODE REVIEW

File: dash/testing/browser.py [hunk 1]
```diff
@@ -247,16 +247,22 @@ def open_new_tab(self, url=None):
...
+                    else webdriver.Remote(
+                        command_executor=remote,
+                        desired_capabilities=getattr(
+                            DesiredCapabilities, self._browser.upper()
+                        ),
+                    )
+                )
+            except WebDriverException:
+                if not i:
+                    raise
```
Comment (alexcjohnson) [inline dash/testing/browser.py]: @byronz I had 3 failures this morning - all from different tests, but with the same message:
```
WebDriverException: Message: unknown error:
Chrome failed to start: exited abnormally
(unknown error: DevToolsActivePort file doesn't exist)
```

It's a bit icky to add a blind retry loop like this, but it seemed to work.

File: tests/integration/test_integration.py [hunk 2]
```diff
@@ -898,3 +896,48 @@ def test_inin022_no_callback_context():
...
+            dcc.Input(id='inner-input')
+        ], id='outer-div'),
+        dcc.Input(id='outer-input')
+    ], id='main')
+
+    ids = ['main', 'inner-div', 'inner-input', 'outer-div', 'outer-input']
+
+    with pytest.raises(NonExistentIdException) as err:
+        @app.callback(
+            Output('nuh-uh', 'children'),
```
Comment (byronz) [inline tests/integration/test_integration.py]: might be slightly better with a multi-output edge case?

File: dash/testing/browser.py (hunk 1 above)
Comment (byronz) [reply dash/testing/browser.py]: I had an impression this crash comes always from one specific python version, like 3.7, or 2.7.   Might be good to figure out the real root cause, but assuming it's just turbulence from the circleci docker. it's more related to bad timing due to load performance of the selenium server.   should we try another approach like the `wait.until`? with a reasonable small wait time like 0.1s, I believe the next try should work (we can give a more tolerant 1s timeout).

Comment (byronz) [reply dash/testing/browser.py]: @alexcjohnson  I created another PR to fix the deprecated message with `pytest.raises`, and I formatted the file with `black`. so if you can format this test_integration file in this PR, we could avoid a lot potential merge conflicts.

Comment (alexcjohnson) [reply dash/testing/browser.py]: I saw this in both 2.7 and 3.7. Feel free to dig in more (preferably in another PR so we can get this bugfix merged), but it's not clear to me what we could wait for, seems like this is happening deep inside Selenium.

File: tests/integration/test_integration.py (hunk 2 above)
Comment (alexcjohnson) [reply tests/integration/test_integration.py]: added multi-output in https://github.com/plotly/dash/pull/821/commits/fbec9dcd866a847e6299d3130bbf8d8cdfce8619

File: dash/testing/browser.py (hunk 1 above)
Comment (byronz) [reply dash/testing/browser.py]: agree, we can leave it now in this PR.
